### PR TITLE
Pract.07 Add 01 | Nikita Beliy

### PR DESCRIPTION
**Pull request by N1k1ta-white**
Nikita Beliy
SI 1 9MI800022

### CODE REVIEW

**Comment by Justsvetoslavov** [comment]
@GeorgiTerziev02 @khatoni Can you take a look?

File: String.cpp [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+const char* String::c_str() const {
+    return str;
+}
+
+bool operator==(const String& str1, const String& str2) {
+    if (str2.size == str1.size)
+        if (strcmp(str1.str, str2.str) == 0)
+            return true;
+    return false;
+}
```
Comment (GeorgiTerziev02) [inline String.cpp:79]: one line

File: String.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    if (str1.size > str2.size)
+        return true;
+    return false;
+}
+
+bool operator>=(const String& str1, const String& str2) {
+    if (str1.size >= str2.size)
+        return true;
+    return false;
+}
```
Comment (GeorgiTerziev02) [inline String.cpp:95]: one line

File: String.cpp [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    res[++size] = '\0';
+    delete str;
+    str = res;
+}
+
+bool String::compare(const String &str) const {
+    if (*this == str)
+        return true;
+    return false;
+}
```
Comment (GeorgiTerziev02) [inline String.cpp:126]: one line, this is opertor==

File: String.cpp [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    return !(str1 > str2);
+}
+
+bool operator<(const String& str1, const String& str2) {
+    return !(str1 >= str2);
+}
+
+unsigned int String::length() const {
+    return size;
+}
```
Comment (GeorgiTerziev02) [inline String.cpp:107]: inconsistent in naming

File: String.cpp [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+String operator+(const String& str1, const String& str2) {
+    String res;
+    res.clear();
+    res.str = new char[str1.size + str2.size + 1];
+    strcpy(res.str, str1.str);
+    strcat(res.str, str2.str);
+    res.size = str1.size + str2.size;
+    return res;
```
Comment (GeorgiTerziev02) [inline String.cpp:57]: create consturctor that preallocates memory. Do not use clear func outside copy constructor and oper=.

File: String.h [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    friend bool operator!=(const String& str1, const String& str2);
+
+    friend bool operator>(const String& str1, const String& str2);
+
+    friend bool operator>=(const String& str1, const String& str2);
+
+    friend bool operator<(const String& str1, const String& str2);
+
+    friend bool operator<=(const String& str1, const String& str2);
+
```
Comment (GeorgiTerziev02) [inline String.h:42]: No sure if u need them all as friend

File: String.h [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    unsigned int length() const;
+
+    bool empty() const;
+
+    void append(const char sym);
+
+    bool compare(const String& str) const;
```
Comment (GeorgiTerziev02) [inline String.h:49]: oper ==

**Review by GeorgiTerziev02 — CHANGES_REQUESTED** [review]

**Comment by GeorgiTerziev02** [comment]
Check the string implementatiton here - [MyString](https://github.com/Angeld55/Object-oriented_programming_FMI/tree/master/Week%2008/String%20and%20StringView)

**Pull request closed without merging**